lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -476,8 +476,16 @@ sub mark_job_linked ($self, $jobid, $referer_url) {
     return undef if !$job || ($referer->path_query =~ /^\/?$/);
     my $comments = $job->comments;
     return undef if $comments->search({text => {like => 'label:linked%'}}, {rows => 1})->first;
+    return undef unless _specific_issue_exists($referer);
     my $user = $self->result_source->schema->resultset('Users')->system({select => ['id']});
     $comments->create_with_event({text => "label:linked Job mentioned in $referer_url", user_id => $user->id});
 }
 
+sub _specific_issue_exists ($url) {
+    my $path = $url->path();
+    my ($issue_id) = $path =~ m/(?:id=|\/)(\d+)/;
```

Comment:
This is wrong in several regards.
1. `$path` is only the path of the url and does not include the `query_string` (the thing after the `?`).

2. If you would do that pattern matching on the path including the `query_string`, then it would still not solve the problem in general.

Please think about the user perspective. We allow to configure **any** bugtracker here. How the id of a certain issue is coded in a url depends on that bugtracker.
It doesn't make sense to let people configure bugtracker hosts, but then have a hardcoded regex to look for a number somewhere specific.